openpype/hosts/blender/plugins/publish/update_current_workfile_last_publish_time.py

```diff
@@ -0,0 +1,20 @@
+import pyblish.api
+import bpy
+
+
+class UpdateCurrentWorkfileLastPublishTime(pyblish.api.ContextPlugin):
+    """Update current workfile last published time.
+
+    Used to ensure current workfile is up to date with published workfiles.
+    """
+
+    order = pyblish.api.IntegratorOrder + 0.00002
+    hosts = ["blender"]
+    families = ["workfile"]
+    label = "Update Current Workfile Last Publish Time"
+    optional = False
+
+    def process(self, context):
+        scene = bpy.context.scene
+        scene["op_published_time"] = context.data["time"]
```

Comment:
Storing into the file itself - doesn't that mean it'd basically become 'incorrect' as soon as someone would save the workfile into another asset (or another task) since the time stored in the scene wouldn't relate to that new context?

Reply:
I guess so but is there a use case for that? I don't think that would be a very good practice, especially as there are other metadatas that would also be incorrect if you did that. I've already seen artists appending workfiles or elements of a workfile and ending up with a broken scene that could not be published (regardless of this feature). I think the same would happen with the scenario you are suggesting.

Reply:
> I guess so but is there a use case for that? I don't think that would be a very good practice, especially as there are other metadatas that would also be incorrect if you did that. I've already seen artists appending workfiles or elements of a workfile and ending up with a broken scene that could not be published (regardless of this feature). I think the same would happen with the scenario you are suggesting.

Could you elaborate on the other type of metadata? We re-use workfiles all the time for other shots. If there are features currently disallowing that I'd say that's a bug and potentially a massive design flaw.

Would love to see reports of issues faced with that.

Reply:
In the current implementation of blender, metadata are stored in instance collections under the `avalon` key, aka `AVALON_PROPERTY`. They contain data such as the family, asset name, subset name, task name and variant.

To have more precise information about whether or not that would be problematic in the current version of openpype, I made a simple workfile with just a model subset containing the default cube. The asset name for this workfile is `Choochoo`. I saved as in another asset (`Pingoo`) folder using the correct syntax for the filename. Then I closed blender and reopened it as that other asset and I checked the metadata of the `modelMain` I created:
```
>>> for key, value in C.active_object.get('avalon').items():
...     print(f"{key}: {value}")
...     
id: pyblish.avalon.instance
family: model
asset: Choochoo
subset: modelMain
active: True
variant: Main
task: Fabrication
```
You can see that my `modelMain` is still recognized as part of the `Choochoo` asset. On one hand this is true. On the other hand, an artist who decides to use this subset as a base for a model in `Pingoo`, could publish its modified model thinking this is fine because they clicked on `Pingoo` earlier, yet the modified model will go in `Choochoo`. And that can be kinda dangerous. So I did it, and while the workfile was published in `Pingoo`, the model was published in `Choochoo`. In this case what should be done is to create a new instance and not publish the one already present in the workfile. In the context of a shot the same could happen with an animation subset for instance.

Does that work differently in other hosts?

All that being said, if you think this should remain doable in blender without confusing the workfile up to date system, I can look into it. Out of the blue I think I could check the asset name to see if the workfile has been moved around and adjust the behavior accordingly.